src/util/base_type.cpp

```diff
@@ -329,6 +338,10 @@ bool base_type_eq(
   return base_type_eq.base_type_eq(type1, type2);
 }
 
+/// Check expressions for equality across all levels of hierarchy.
```

Comment:
Sorry I don't understand the meaning of this - perhaps a short example as you've given for the `typet` overload